clients/drcachesim/tracer/output.cpp

```diff
@@ -1005,13 +1008,30 @@ process_and_output_buffer(void *drcontext, bool skip_size_cap)
 
     if (do_write) {
         bool hit_window_end = false;
-        if (op_trace_for_instrs.get_value() > 0) {
+
+        if (op_L0_filter_until_instrs.get_value() && mode == BBDUP_MODE_L0_FILTER) {
+            uintptr_t toadd =
+                *(uintptr_t *)TLS_SLOT(data->seg_base, MEMTRACE_TLS_OFFS_ICOUNT);
+            hit_window_end = count_traced_instrs(drcontext, toadd,
```

Comment:
It's confusing to use hit_window_end for storing whether we exceeded op_L0_filter_until_instrs. We should rename it to something like reached_L0_filter_until_instrs_limit.

Also, looks like hit_window_end is currently declared in a larger scope than it needs to be. We can declare two variables local to this if-block and the next else-if block, and name them both specifically.

Reply:
Done.